### PR TITLE
Editorial suggestions from Benjamin Kaduk (invariants, TLS)

### PR DESCRIPTION
**Pull request by kaduk**
I made some proposed tweaks in my local copy as I was doing my IESG review.
This pull request contains the suggetsions for -invariants and -tls (sorry if that is not the preferred workflow; there's just one change to invariants and it's in its own commit, so it will be easy to separate if needed).

### CODE REVIEW

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -720,10 +723,15 @@ notably those caused by the potential exposure to replay attack; see {{replay}}.
...
-max_early_data_size of 0xffffffff.  The TLS max_early_data_size parameter is not
-used in QUIC.  The amount of data that the client can send in 0-RTT is
+The TLS "early_data" extension in the NewSessionTicket message is defined
+to convey (in the "max_early_data_size" parameter) the amount of TLS 0-RTT
+data the server is willing to accept.  QUIC does not use TLS 0-RTT data;
+instead, QUIC has its own 0-RTT stream.  Accordingly, the
+"max_early_data_size" parameter is repurposed to hold a sentinel value
+0xffffffff to indicate that the server is willing to accept QUIC 0-RTT data;
+to indicate that the server does not accept 0-RTT data, the "early_data"
+extension is omitted from the NewSessionTicket.
```
Comment (kaduk) [inline draft-ietf-quic-tls.md:733]: (This all might become moot depending on the resolution of my discuss point.)

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -154,7 +154,7 @@ Internally, TLS is a layered protocol, with the structure shown in
 
 ~~~~
           +-------------+------------+--------------+---------+
-Handshake |             |            |  Application |         |
+Content   |             |            |  Application |         |
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply draft-ietf-quic-tls.md:157]: @ekr, do you have any opinion on this?

File: draft-ietf-quic-tls.md [hunk 3]
```diff
@@ -403,8 +406,8 @@ acquires handshake bytes before sending its first packet.  A QUIC server starts
 the process by providing TLS with the client's handshake bytes.
 
 At any time, the TLS stack at an endpoint will have a current sending
-encryption level and receiving encryption level. Encryption levels determine
-the packet type and keys that are used for protecting data.
+encryption level and receiving encryption level. TLS Encryption levels determine
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:409]:
```suggestion
encryption level and receiving encryption level. TLS encryption levels determine
```

File: draft-ietf-quic-tls.md [hunk 4]
```diff
@@ -720,10 +723,15 @@ notably those caused by the potential exposure to replay attack; see {{replay}}.
...
 ### Enabling 0-RTT {#enable-0rtt}
 
-To communicate their willingness to process 0-RTT data, servers send a
-NewSessionTicket message that contains the early_data extension with a
-max_early_data_size of 0xffffffff.  The TLS max_early_data_size parameter is not
-used in QUIC.  The amount of data that the client can send in 0-RTT is
+The TLS "early_data" extension in the NewSessionTicket message is defined
+to convey (in the "max_early_data_size" parameter) the amount of TLS 0-RTT
+data the server is willing to accept.  QUIC does not use TLS 0-RTT data;
+instead, QUIC has its own 0-RTT stream.  Accordingly, the
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:729]:
```suggestion
data the server is willing to accept.  QUIC does not use TLS 0-RTT data.
QUIC uses 0-RTT packets to carry early data.  Accordingly, the
```

File: draft-ietf-quic-tls.md [hunk 5]
```diff
@@ -2142,9 +2151,9 @@ Because the header uses a 4-byte packet number encoding, the first 16 bytes of
 the protected payload is sampled, then applied to the header:
 
 ~~~
-sample = d1b1c98dd7689fb8ec11d242b123dc9b
+sampled_data = d1b1c98dd7689fb8ec11d242b123dc9b
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:2154]:
```suggestion
sampled_bytes = d1b1c98dd7689fb8ec11d242b123dc9b
```

File: draft-ietf-quic-tls.md [hunk 6]
```diff
@@ -2142,9 +2151,9 @@ Because the header uses a 4-byte packet number encoding, the first 16 bytes of
...
 
-mask = AES-ECB(hp, sample)[0..4]
+mask = AES-ECB(hp, sampled_data)[0..4]
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:2156]:
```suggestion
mask = AES-ECB(hp, sampled_bytes)[0..4]
```

File: draft-ietf-quic-tls.md [hunk 7]
```diff
@@ -2050,7 +2059,7 @@ included. All values are shown in hexadecimal.
 
 ## Keys
 
-The labels generated during the execution of the HKDF-Expand-Label function and
+The labels (Hkdf.label) generated during the execution of the HKDF-Expand-Label function and
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:2062]: I don't think that this is right.  I think that you mean:

```suggestion
The labels generated during the execution of the HKDF-Expand-Label function (that is, HkdfLabel.label) and part of the value
```

File: draft-ietf-quic-tls.md [hunk 8]
```diff
@@ -2221,7 +2230,7 @@ As a result, after protection, the header protection sample is taken starting
 from the third protected byte:
 
 ~~~
-sample = 2cd0991cd25b0aac406a5816b6394100
+sampled_data = 2cd0991cd25b0aac406a5816b6394100
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:2233]:
```suggestion
sampled_bytes = 2cd0991cd25b0aac406a5816b6394100
```

File: draft-ietf-quic-tls.md [hunk 9]
```diff
@@ -2300,7 +2309,7 @@ The resulting ciphertext is the minimum size possible. One byte is skipped to
 produce the sample for header protection.
 
 ~~~
-sample = 5e5cd55c41f69080575d7999c25a5bfb
+sampled_data = 5e5cd55c41f69080575d7999c25a5bfb
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md:2312]:
```suggestion
sampled_bytes = 5e5cd55c41f69080575d7999c25a5bfb
```

**Review by martinthomson — APPROVED** [review]
Thanks Ben, this is mostly good.  I have a few suggestions, but I am planning to accept this with those (and some cleanup) assuming that we don't get better advice on "Content Layer".

File: draft-ietf-quic-tls.md [hunk 10]
```diff
@@ -681,9 +684,9 @@ Endpoints that use session resumption might need to remember some information
 about the current connection when creating a resumed connection. TLS requires
 that some information be retained; see Section 4.6.1 of {{!TLS13}}. QUIC itself
 does not depend on any state being retained when resuming a connection, unless
-0-RTT is also used; see {{enable-0rtt}} and Section 7.4.1 of
```
[the remark this answers is not in the document]
Comment (MikeBishop) [reply draft-ietf-quic-tls.md:684]: This might even warrant an "...of this document" to make it explicit.

File: draft-ietf-quic-tls.md (hunk 7 above)
Comment (kaduk) [reply draft-ietf-quic-tls.md:2062]: Oops, yes, the HkdfLabel bit got garbled in transmission from brain to fingers.
Your suggestion looks fine; I just wanted to reiterate that it's the whole HkdfLabel.label (including the "tls13 " prefix) that we show, not just the argument to HKDF-Expand-Label.

**Comment by janaiyengar** [comment]
At a high level, I'm ok with the editorial changes, but I'd like to push back against the changes to variable and field names at a minimum. We've had a lot of eyes go over this so far, and changing names risks introducing inconsistencies or omissions that we can't afford to do at this late stage.

**Review by janaiyengar — CHANGES_REQUESTED** [review]
At a high level, I'm ok with the editorial changes, but I'd like to push back against the changes to variable and field names at a minimum. We've had a lot of eyes go over this so far, and changing names risks introducing inconsistencies or omissions that we can't afford to do at this late stage.

**Comment by kaduk** [comment]
> At a high level, I'm ok with the editorial changes, but I'd like to push back against the changes to variable and field names at a minimum. We've had a lot of eyes go over this so far, and changing names risks introducing inconsistencies or omissions that we can't afford to do at this late stage.

Just to check: this is basically just the `sample`->`sampled_bytes` change?
The immediate impetus for that proposal is from the way this chunk renders in the text version:

```
   Header protection uses the output of the packet
   protection AEAD to derive "sample", and then encrypts the header
   field using a pseudorandom function (PRF) as follows:
```

In the HTML and PDF formats there's different formatting, but this one makes it look like the quotes are surrounding the literal output of the derivation process, as opposed to a name for the output of the derivation process. 

That issue could probably be avoided with just a local rewording in this one instance (though I do think that "sampled_bytes" is an objectively better name for what it is).

**Comment by janaiyengar** [comment]
> Just to check: this is basically just the sample->sampled_bytes change?
Yes, that and the Content Layer changes. My contention is that we're unlikely to get a lot of eyes over these documents, and I am objecting to cosmetic changes that have high risk at this late stage.

**Comment by kaduk** [comment]
> > Just to check: this is basically just the sample->sampled_bytes change?
> 
> Yes, that and the Content Layer changes. My contention is that we're unlikely to get a lot of eyes over these documents, and I am objecting to cosmetic changes that have high risk at this late stage.

Hmm, not-Handshake-Layer is basically the thing I care about the most in this PR.  That specific discussion seemed quite localized.  (The word "layer" only appears 16 times in the document, which seems feasible to exhaustively check for a "Handshake" prefix.)

**Comment by kaduk** [comment]
I reverted the `sampled_bytes`/`sampled_data` stuff and proposed an alternative local rewording in the prose that doesn't affect the variable name or pseudocode.

I did not revert the "Handshake Layer"-->"Content Layer" change, and would still like to hear what @ekr thinks about that topic.

**Pull request merged**